go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -1037,6 +1037,10 @@ func (tm *TabletManager) StopReplicationAndGetStatus(ctx context.Context, stopRe
 	before := replication.ReplicationStatusToProto(rs)
 	before.BackupRunning = tm.IsBackupRunning()
 
+	// Get semi-sync state before replication is stopped.
+	before.SemiSyncPrimaryEnabled, before.SemiSyncReplicaEnabled = tm.MysqlDaemon.SemiSyncEnabled(ctx)
+	before.SemiSyncPrimaryStatus, before.SemiSyncReplicaStatus = tm.MysqlDaemon.SemiSyncStatus(ctx)
```

Comment:
It would be nice to skip this on async replicas. In earlier commits I passed a "capability" to this RPC to achieve this, but I later realised we don't avoid simple `SELECT`s in many places, so it felt like overkill. This could be reintroduced, however